modules/imgcodecs/src/loadsave.cpp

```diff
@@ -50,10 +50,55 @@
 #undef min
 #undef max
 #include <iostream>
+#include <fstream>
 
 /****************************************************************************************\
 *                                      Image Codecs                                      *
 \****************************************************************************************/
+namespace {
+
+class ByteStreamBuffer : public std::streambuf
+{
+public:
+   ByteStreamBuffer(char* base, size_t length)
+   {
+      setg( base, base, base + length );
+   };
+
+protected:
+   virtual pos_type seekoff( off_type off,
+                             std::ios_base::seekdir dir,
+                             std::ios_base::openmode )
+   {
+      if( off > INT_MAX )
+      {
+         return -1;
+      }
+
+      if( off < INT_MIN )
+      {
+         return -1;
+      }
+
+      if( dir == std::ios_base::cur )
+      {
+         if( gptr() + off >= egptr() )
+         {
+            return -1;
+         }
+         if( gptr() + off < eback() )
+         {
+            return -1;
+         }
+         gbump( static_cast<int>(off) );
```

Comment:
it can't be dropped because the windows build complains about silently converting a long to an int